docs/src/howToGetStarted.md

```diff
@@ -38,105 +281,267 @@ for f in dataFiles
 end
 ```
 
-Read meta-data and panel as a `DataFrame`, and make sure that the column names match the CyTOF
-FCS file names:
+The metadata is present in external files; we read it into a DataFrame and
+extract information about FCS data columns from there:
 
 ```julia
-# Read  files as DataFrames
+using GigaSOM
+
+# Read the dataset description in XLSX files as DataFrames
 md = GigaSOM.DataFrame(GigaSOM.XLSX.readtable("PBMC8_metadata.xlsx", "Sheet1")...)
 panel = GigaSOM.DataFrame(GigaSOM.XLSX.readtable("PBMC8_panel.xlsx", "Sheet1")...)
-panel[:Isotope] = map(string, panel[:Isotope])
-panel[:Metal] = map(string, panel[:Metal])
-panel[:Antigen] = map(string, panel[:Antigen])
-panel.Metal[1]=""
-GigaSOM.insertcols!(panel,4,:fcs_colname => map((x,y,z)->x.*"(".*y.*z.*")".*"Dd",panel[:Antigen],panel[:Metal],panel[:Isotope]))
-print(panel.fcs_colname)
+
+# Get the parameters structure from the first FCS files
+_, fcsParams = loadFCSHeader(md[1, :file_name])
+
+# See what antigens are saved in there
+_, fcsAntigens = getMarkerNames(getMetaData(fcsParams))
+
+# See which antigens we want to use (assume we want only the lineage markers
+antigens = panel[panel[:,:Lineage].==1, :Antigen]
+
+# Make the names a bit more Julia-friendly and predictable:
+cleanNames!(antigens)
+cleanNames!(fcsAntigens)
 ```
 
-Extract the lineage and functional markers with `getMarkers()` function:
+### Loading and preparing the data
+
+Now we have the vector of `fcsAntigens` that the FCS files store, and list of
+`antigens` that we want to analyze. We continue by loading the data, reducing
+it to the desired antigens and transforming it a bit:
 
 ```julia
-lineageMarkers, functionalMarkers = getMarkers(panel)
+di = loadFCSSet(:pbmc8, md[:,:file_name])
 ```
 
-Read FCS files `readFlowset()`:
+(If data distribution and parallelization is required, you must add parallel
+workers using `addprocs` **before** this step.)
 
+Now that the data is loaded, let's prepare them a bit by reducing to actual
+interesting columns, transformation and scaling:
 
 ```julia
-fcsRaw = readFlowset(md.file_name)
-```
+# select only the antigens that correspond to lineage markers (as prepared above)
+dselect(di, fcsAntigens, antigens)
+
+cols=Vector(1:length(antigens)) # shortcut for "all rows"
+
+# perform asinh transformation on all data in the dataset, '5' is the cofactor
+dtransform_asinh(di, cols, 5)
 
-`readFlowset()` is a wrapper function around [FCSFiles.jl](https://github.com/tlnagy/FCSFiles.jl). Please note the current limitations
-of this package (i.e., the [limit for large files](https://github.com/tlnagy/FCSFiles.jl/blob/master/src/parse.jl#L20)).
+# normalize all dataset columns to mean=0 sdev=1
+dscale(di, cols)
+```
 
-Clean names to remove problematic characters in the column names:
+### Creating a Self Organizing MAP (SOM)
 
+With the data prepared, running the SOM algorithm is almost the same as in
```

Comment:
This is a bit repetition of above. Maybe split this case up into a separate file.